gwml2_schema.sql

```diff
@@ -0,0 +1,102 @@
+CREATE SCHEMA groundwater;
+
+CREATE TABLE groundwater.CI_Telephone (
+	id serial PRIMARY KEY,
+	voice VARCHAR(50),
+	facsimile VARCHAR(50)
+);
+
+CREATE TABLE groundwater.CI_Address (
+	id serial PRIMARY KEY,
+	deliveryPoint VARCHAR(50),
+	city VARCHAR(150),
+	administrativeArea VARCHAR(150),
+	postalCode VARCHAR(50),
+	country VARCHAR(150),
+	electronicMailAddress VARCHAR(150)
+);
+
+CREATE TABLE groundwater.CI_OnlineResource (
+	id serial PRIMARY KEY,
+	linkage VARCHAR(150),
+	protocol VARCHAR(150),
+	applicationProfile VARCHAR(250),
+	name VARCHAR(150),
+	description VARCHAR(255),
+	function groundwater.CI_OnLineFunctionTerm
+);
+
+CREATE TABLE groundwater.CI_Contact (
+	id serial PRIMARY KEY,
+	phone int4 REFERENCES groundwater.CI_Telephone(id) ON DELETE SET NULL,
+	address int4 REFERENCES groundwater.CI_Address(id) ON DELETE SET NULL,
+	onlineResource int4 REFERENCES groundwater.CI_OnlineResource(id) ON DELETE SET NULL,
+	hoursOfService VARCHAR (255),
+	contactInstructions VARCHAR (255)
+);
+
+CREATE TABLE groundwater.CI_ResponsibleParty (
+	id serial PRIMARY KEY,
+	individualName VARCHAR (255),
+	organisationName VARCHAR (255),
+	positionName VARCHAR (255),
+	contactInfo int4 REFERENCES groundwater.CI_Contact(id) ON DELETE SET NULL,
+	role groundwater.CI_RoleTerm
+);
+
+CREATE TABLE groundwater.GM_Envelope (
+	id serial PRIMARY KEY,
+	upperCorner GEOMETRY,
+	lowerCorner GEOMETRY
```

Comment:
add a virtual column later that turns this into a rectangle geometry

Reply:
Also there should be a grometry constraint specifying that the geometry is a point

Reply:
> add a virtual column later that turns this into a rectangle geometry

Actually reading the spec further I think we should not have the virtual column. The envelope in this case references a section in the vertical segment of a borehole, so as I understand it we should have a virtual column that converst start - end into a line segment with 3d geometry. Make sure your point geometry is constrained to be a point 3d.